ci/docker/data12.build.wanda.yaml

```diff
@@ -10,6 +10,6 @@ srcs:
   - python/requirements/ml/data-requirements.txt
   - python/requirements/ml/data-test-requirements.txt
 build_args:
-  - ARROW_VERSION=12
+  - ARROW_VERSION=12.*
```

Comment:
this config is now the same as https://github.com/ray-project/ray/blob/master/ci/docker/datab.build.wanda.yaml#L13; do you mind also help delete that file + remove the corresponding test blob in https://github.com/ray-project/ray/blob/master/.buildkite/data.rayci.yml#L42-L53, thankkks